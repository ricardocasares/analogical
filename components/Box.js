import React from "react";
import { WHITE, BLACK, LIGHT_GRAY } from "../lib/colors";

export default ({ children }) => (
  <div className="box">
    {children}
    <style jsx>{`
      .box {
        border: 1px solid ${LIGHT_GRAY};
        border-radius: 3px;
        padding: 20px;
        margin-bottom: 20px;
        margin-right: 20px;
        width: 100%;
        transition: all 0.2s;
        position: relative;
      }

      .box:last-of-type {
        margin-right: 0;
      }

      .box:hover {
        transform: translateY(-9px);
        box-shadow: 0 2px ${WHITE}, 0 3px ${LIGHT_GRAY}, 0 5px ${WHITE},
          0 6px ${LIGHT_GRAY}, 0 8px ${WHITE}, 0 9px ${LIGHT_GRAY};
      }
    `}</style>
  </div>
);
