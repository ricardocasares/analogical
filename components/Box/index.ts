import styled from "@app/lib/styled";
import {
  alignItems,
  AlignItemsProps,
  borders,
  BorderProps,
  boxShadow,
  BoxShadowProps,
  color,
  ColorProps,
  display,
  DisplayProps,
  maxHeight,
  MaxHeightProps,
  minHeight,
  MinHeightProps,
  maxWidth,
  MaxWidthProps,
  minWidth,
  MinWidthProps,
  ratio,
  RatioProps,
  space,
  SpaceProps,
  size,
  SizeProps,
  flex,
  FlexProps,
  flexDirection,
  FlexDirectionProps,
  position,
  PositionProps
} from "styled-system";

export type Box = AlignItemsProps &
  BorderProps &
  BoxShadowProps &
  ColorProps &
  DisplayProps &
  MaxHeightProps &
  MinHeightProps &
  MaxWidthProps &
  MinWidthProps &
  RatioProps &
  SpaceProps &
  SizeProps &
  FlexProps &
  FlexDirectionProps &
  PositionProps;

// @ts-ignore
export const Box = styled.div<Box>`
  ${alignItems}
  ${borders}
  ${boxShadow}
  ${color}
  ${display}
  ${flex}
  ${flexDirection}
  ${maxHeight}
  ${minHeight}
  ${maxWidth}
  ${minWidth}
  ${ratio}
  ${size}
  ${space}
  ${position}
`;

export default Box;
