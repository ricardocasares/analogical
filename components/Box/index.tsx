import styled from "@app/lib/styled";
import {
  alignItems,
  AlignItemsProps,
  border,
  BorderProps,
  borderColor,
  BorderColorProps,
  borderRadius,
  BorderRadiusProps,
  boxShadow,
  BoxShadowProps,
  color,
  ColorProps,
  display,
  DisplayProps,
  maxHeight,
  MaxHeightProps,
  minHeight,
  MinHeightProps,
  maxWidth,
  MaxWidthProps,
  minWidth,
  MinWidthProps,
  ratio,
  RatioProps,
  space,
  SpaceProps,
  size,
  SizeProps,
  flex,
  FlexProps,
  flexDirection,
  FlexDirectionProps,
  flexWrap,
  FlexWrapProps,
  gridGap,
  GridGapProps,
  gridTemplateColumns,
  GridTemplatesColumnsProps,
  gridTemplateRows,
  GridTemplatesRowsProps,
  position,
  PositionProps
} from "styled-system";

export type Box = AlignItemsProps &
  BorderProps &
  BorderColorProps &
  BorderRadiusProps &
  BoxShadowProps &
  ColorProps &
  DisplayProps &
  MaxHeightProps &
  MinHeightProps &
  MaxWidthProps &
  MinWidthProps &
  RatioProps &
  SpaceProps &
  SizeProps &
  FlexProps &
  FlexDirectionProps &
  FlexWrapProps &
  GridGapProps &
  GridTemplatesColumnsProps &
  GridTemplatesRowsProps &
  PositionProps & {
    as?: string;
  };

// @ts-ignore
export const Box = styled.div<Box>`
  ${alignItems}
  ${border}
  ${borderColor}
  ${borderRadius}
  ${boxShadow}
  ${color}
  ${display}
  ${flex}
  ${flexDirection}
  ${flexWrap}
  ${gridGap}
  ${gridTemplateColumns}
  ${gridTemplateRows}
  ${maxHeight}
  ${minHeight}
  ${maxWidth}
  ${minWidth}
  ${ratio}
  ${size}
  ${space}
  ${position}
`;

export default Box;
