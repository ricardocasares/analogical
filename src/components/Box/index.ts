import styled from "@emotion/styled";
import { As } from "@/types";
import {
  flexbox,
  FlexboxProps,
  background,
  BackgroundProps,
  border,
  BorderProps,
  compose,
  space,
  SpaceProps,
  layout,
  LayoutProps,
} from "styled-system";

export type Box = As &
  FlexboxProps &
  LayoutProps &
  BackgroundProps &
  BorderProps &
  SpaceProps;

export const Box = styled.div<Box>(
  compose(flexbox, layout, background, border, space)
);
