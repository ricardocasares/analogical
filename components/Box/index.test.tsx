import React from "react";
import renderer from "react-test-renderer";
import serializer from "jest-emotion";
import Box from "@app/components/Box";

expect.addSnapshotSerializer(serializer);

test("renders with correct styles", () => {
  expect(
    renderer.create(<Box p={[1, 2]}>hello world</Box>).toJSON()
  ).toMatchSnapshot();
});
