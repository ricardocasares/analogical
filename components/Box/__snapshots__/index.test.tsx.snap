// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with correct styles 1`] = `
.emotion-0 {
  padding: 4px;
}

@media screen and (min-width:40em) {
  .emotion-0 {
    padding: 8px;
  }
}

<div
  className="emotion-0"
>
  hello world
</div>
`;
